chainer/functions/loss/decov.py

```diff
@@ -31,32 +36,56 @@ def forward(self, inputs):
         self.covariance = self.h_centered.T.dot(self.h_centered)
         xp.fill_diagonal(self.covariance, 0.0)
         self.covariance /= len(h)
-        cost = xp.vdot(self.covariance, self.covariance) * h.dtype.type(0.5)
-        return utils.force_array(cost),
+        if self.reduce == 'half_frobenius_norm':
+            cost = xp.vdot(self.covariance, self.covariance)
```

Comment:
Ah, yes. I think it is better to provide squared norm (like MSE). So, I change the option to `'half_squared_frobenius_norm'`.